### PR TITLE
doctor: Run cask doctor checks in brew doctor

### PR DESCRIPTION
**Pull request by whoiswillma**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Move cask doctor checks into `Diagnostics::Checks`. If the user has casks installed, running brew doctor will also run cask-related checks. 

brew doctor is normally silent unless a warning arises, but `brew cask doctor` outputs a whole bunch of diagnostic information. To preserve the quietness of brew doctor, but keep the verboseness of `brew cask doctor`, I added a flag to `Diagnostic::Checks` that toggles additional info. This can be toggled on for brew doctor using `--verbose`.

### CODE REVIEW

**Comment by jonchang** [comment]
```
==> brew doctor
==> FAILED
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: The staging path /usr/local/Caskroom is not writable by the current user.
To fix, run 'sudo chown -R $(whoami):staff /usr/local/Caskroom'
```

This is not writeable by the self-hosted runner user for security reasons. Can you add something to skip this check when we're on CI, e.g. `return if ENV["HOMEBREW_GITHUB_ACTIONS"]` and add a comment to that effect?

File: Library/Homebrew/diagnostic.rb [hunk 1]
```diff
@@ -854,9 +873,163 @@ def check_deleted_formula
...
+      def check_cask_software_versions
+        add_info "Homebrew Version", HOMEBREW_VERSION
+        add_info "macOS", MacOS.full_version
+        add_info "SIP", begin
+          csrutil = "/usr/bin/csrutil"
+          if File.executable?(csrutil)
+            Open3.capture2(csrutil, "status")
+                 .first
+                 .gsub("This is an unsupported configuration, likely to break in " \
+                   "the future and leave your machine in an unknown state.", "")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/diagnostic.rb:885]:
```suggestion
                 .gsub("This is an unsupported configuration, likely to break in " \
                       "the future and leave your machine in an unknown state.", "")
```

File: Library/Homebrew/diagnostic.rb [hunk 2]
```diff
@@ -854,9 +873,163 @@ def check_deleted_formula
...
+          else
+            "N/A"
+          end
+        end
+        add_info "Java", SystemConfig.describe_java
+
+        nil
+      end
+
+      # This could be done by calling into Homebrew, but the situation
```
Comment (MikeMcQuaid) [inline Library/Homebrew/diagnostic.rb:899]: This comment no longer makes much sense and can be removed.

**Comment by MikeMcQuaid** [comment]
Thanks for this, great work again @whoiswillma!

**Pull request merged**